### ISSUE TITLE
Help getting NUC123 in bootloader mode

### ISSUE THREAD

**Issue by fufexan** [elfmimi/nu-isp-rs#1]
I'm trying to flash my Ducky One 2 Mini keyboard using the instructions [here](https://github.com/qmk/qmk_firmware/pull/11112), but I can't seem to get it in bootloader mode. I'm using Linux (NixOS if it matters).

I manage to get the board to freeze if I run `nu-isp-cli 0416:0123 info`, but trying to flash after doing that doesn't do anything, and replugging the board makes it work like before.

Sometimes I get this error
```sh
$ nu-isp-cli 0416:0123 info
Printing info...
hidapi error: hid_error is not implemented yet!
ERROR! HidError(HidApiError { message: "hid_error is not implemented yet" })
```
or this one
```sh
$ nu-isp-cli 0416:0123 info
Printing info...
 ERROR nu_isp_cli > nu-isp-cli/src/main.rs(423): Failed opening hid device
Are you sure device is plugged in and in bootloader mode?
```
Is there some library I've left out while building the program?

**Comment by elfmimi** [comment]
Maybe the true reason is as stated in this line.
>  ERROR nu_isp_cli > nu-isp-cli/src/main.rs(423): Failed opening hid device
That means you don't have the right permission to open the device file.

Try adding a udev rule like this.
> SUBSYSTEM=="usb", ATTR{idVendor}=="0416", ATTR{idProduct}=="0123", MODE="666"
or
Run it with root privilege.

I'll change this message to make it easy to spot that you need permission.
> Are you sure device is plugged in and in bootloader mode?

**Comment by elfmimi** [comment]
nu-isp-cli is designed with no specific hardware in mind. so technically this issue is off-topic.

0416:0123 is the VID:PID pair for when Ducky One2Mini is functioning as a keyboard. specifying that in the command-line is clearly wrong operation. nu-isp-cli do not come with means to put devices into boot-loader mode.

You have to manually put your keyboard in to boot-loader mode somehow.
The instruction says:
> For the 1861ST to flash you can boot the keyboard while holding D+L.

Here "boot" is referring to the action powering your keyboard, in other words the action connecting a USB cable to it.

Then your keyboard in boot-loader mode should show up as 0416:3F00 .

In case of Linux like system, be aware of permissions of the device file. that is, take care of udev rules OR run with root privilege.

**Comment by fufexan** [comment]
Thanks for the response.

I've already added the relevant udev rules like specified in the readme, and the only thing I seem to be missing is getting the board into bootloader mode.

**Comment by runfalk** [comment]
Might be a bit late, but I get my Ducky One 2 Mini RGB (dkon2061st) into bootloader (ISP mode) by holding D while I plug in the USB cable. You know it's in bootloader mode since the backlight isn't running and the keys aren't working.
